lib/olive_branch/middleware.rb

```diff
@@ -14,18 +14,20 @@ def call(env)
       status, headers, response = @app.call(env)
 
       if inflection && headers["Content-Type"] =~ /application\/json/
-        new_response = JSON.parse(response.body)
+        response.each do |body|
+          new_response = JSON.parse(body)
 
-        if inflection == "camel"
-          new_response.deep_transform_keys! { |k| k.camelize(:lower) }
-        elsif inflection == "dash"
-          new_response.deep_transform_keys!(&:dasherize)
-        end
+          if inflection == "camel"
+            new_response.deep_transform_keys! { |k| k.camelize(:lower) }
+          elsif inflection == "dash"
+            new_response.deep_transform_keys!(&:dasherize)
+          end
 
-        [status, headers, [new_response.to_json]]
-      else
-        [status, headers, response]
+          body.replace(new_response.to_json)
```

Comment:
`.replace` is just a thing you can do to strings: http://ruby-doc.org/core-1.9.3/String.html#method-i-replace

Never had much use for it before now, but I needed some way to manipulate the response, and `ActionDispatch::Response::RackBody` doesn't give you many (any?) other options.

According to the Rack website (http://rack.github.io/), the response body "must respond to `each`", so this seemed like the most universal approach.